src/app/blocks/grid-block/grid-block.component.ts

```diff
@@ -64,32 +57,28 @@ export class GridBlockComponent implements OnInit, OnChanges {
           if (this.firstRowHeaders) column.headerName = this.model[0][key];
           return column;
         }        
-      )
-      if (this.firstRowHeaders) this.model.shift(); // remove first row
+      )      
     }
+    let workingModel = this.model; // create a local clone of the data - to allow us to modify data only for display
+    if (this.firstRowHeaders && workingModel.length >0) workingModel.shift(); // remove first row from the working model
```

Comment:
I am now cloning the model to a local var. Original data is untouched.

Reply:
That makes sense but there is an actual clone method BTW that lodash has, which is already imported (and used below).

Reply:
Now using "clone". Not just pretending.